### PR TITLE
Price variation fixes

### PR DESCRIPTION
**Pull request by hannunosto**
* Add helper for resolving variation id (when using price variations)
* fix serialization
* add default variation id for the account

### CODE REVIEW

File: app/code/community/Nosto/Tagging/Model/Meta/Variation/Collection.php [hunk 1]
```diff
@@ -52,26 +52,22 @@ public function loadData(
...
-        }
-
-        $groups = Mage::getModel('customer/group')->getCollection();
-        /** @var Mage_Customer_Model_Group $group */
-        foreach ($groups as $group) {
-            //skip the default customer group
-            if ($group->getCode() == $defaultGroupCode) {
-                continue;
+        $groups = $product->getData('group_price');
+        if (is_array($groups)) {
```
Comment (chenliangde) [inline app/code/community/Nosto/Tagging/Model/Meta/Variation/Collection.php]: Catalog price rule is not taking into considering.

File: app/code/community/Nosto/Tagging/Model/Meta/Variation/Collection.php [hunk 2]
```diff
@@ -52,26 +52,20 @@ public function loadData(
...
+            if ($group->getId() == Nosto_Tagging_Helper_Variation::DEFAULT_CUSTOMER_GROUP_ID) {
                 continue;
             }
-
-            /** @var Nosto_Tagging_Model_Meta_Variation $variation */
-            $variation = Mage::getModel('nosto_tagging/meta_variation');
-            $variation->loadData($product, $group, $productAvailability, $currencyCode, $store);
-            $this->append($variation);
+            if ($group instanceof Mage_Customer_Model_Group) {
+                /** @var Nosto_Tagging_Model_Meta_Variation $variation */
```
Comment (chenliangde) [inline app/code/community/Nosto/Tagging/Model/Meta/Variation/Collection.php]: If the null check is needed, it should be checked before the line 60

File: app/code/community/Nosto/Tagging/Helper/Variation.php [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+ * @category Nosto
+ * @package  Nosto_Tagging
+ * @author   Nosto Solutions Ltd <magento@nosto.com>
+ */
+class Nosto_Tagging_Helper_Variation extends Mage_Core_Helper_Abstract
+{
+    /**
+     * Default customer group id for generating the variation_id
+     */
+    const DEFAULT_CUSTOMER_GROUP_ID = 0;
```
Comment (chenliangde) [inline app/code/community/Nosto/Tagging/Helper/Variation.php]: How about: const DEFAULT_CUSTOMER_GROUP_ID = Mage_Customer_Model_Group::NOT_LOGGED_IN_ID;

File: app/code/community/Nosto/Tagging/Model/Meta/Product.php [hunk 4]
```diff
@@ -116,8 +117,6 @@ public function loadData(Mage_Catalog_Model_Product $product, Mage_Core_Model_St
         if (($tags = $this->buildTags($product, $store)) !== array()) {
             $this->setTag1($tags);
         }
-
-        $this->amendPrice($product, $store);
         $this->amendAttributeTags($product, $store);
```
Comment (chenliangde) [inline app/code/community/Nosto/Tagging/Model/Meta/Product.php]: I would keep this method. Even though the result is the same currently, but it is better to do the right thing here. So in the future we could focus on the "getting not-logged-in price" issue.

File: app/code/community/Nosto/Tagging/Model/Meta/Product.php [hunk 5]
```diff
@@ -305,25 +299,6 @@ protected function amendAlternativeImages(
     }
 
     /**
-     * Amends the product price to the price that set for default customer group.
-     *
-     * @param Mage_Catalog_Model_Product $product the product model.
-     * @param Mage_Core_Model_Store $store the store model.
-     *
-     */
-    protected function amendPrice(Mage_Catalog_Model_Product $product, Mage_Core_Model_Store $store)
```
Comment (chenliangde) [inline app/code/community/Nosto/Tagging/Model/Meta/Product.php]: I would keep this method and modify it to use "DEFAULT_CUSTOMER_GROUP_ID". Even though the result is the same currently, but it is better to do the right thing here. So in the future we could focus on the "getting not-logged-in price" issue.

File: app/code/community/Nosto/Tagging/Helper/Variation.php (hunk 3 above)
Comment (chenliangde) [inline app/code/community/Nosto/Tagging/Helper/Variation.php]: And the name is a bit confusing with magento's default customer group. How about use Mage_Customer_Model_Group::NOT_LOGGED_IN_ID directly?

Comment (hannunosto) [inline app/code/community/Nosto/Tagging/Helper/Variation.php]: The idea to use own constant for this is that we can control what is the default customer group id for price variation. If we use directly `Mage_Customer_Model_Group::NOT_LOGGED_IN_ID` we don't have that control.

File: app/code/community/Nosto/Tagging/Model/Meta/Product.php (hunk 5 above)
Comment (hannunosto) [inline app/code/community/Nosto/Tagging/Model/Meta/Product.php]: This adds yet another db query and we want to keep the amount if queries minimal

**Pull request merged**